### PR TITLE
add storeLabels setting

### PR DESCRIPTION
**Pull request by xiaojingchen**
### What problem does this PR solve? 
- add storeLabels setting
### Check List 

Tests 
 - E2E test
 - Stability test
 - Manual test [done]

Code changes

 - Has Helm charts change
 - Has Go code change

Side effects
user must know current storeLabel cannot support contains the slash character.

Related changes
 - Need to cherry-pick to the release branch
 - Need to update the documentation

### Does this PR introduce a user-facing change?:
 
 ```release-note
Add `tikv.storeLables` setting
 ```

### CODE REVIEW

File: charts/tidb-cluster/values.yaml [hunk 1]
```diff
@@ -47,6 +47,7 @@ discovery:
 # if the ConfigMap was not changed.
 enableConfigMapRollout: false
 
+
```
Comment (weekface) [inline charts/tidb-cluster/values.yaml]: remove this line?

File: charts/tidb-cluster/templates/tidb-cluster.yaml [hunk 2]
```diff
@@ -67,6 +67,16 @@ spec:
     annotations:
 {{ toYaml .Values.tikv.annotations | indent 6 }}
   {{- end }}
+  {{- if .Values.tikv.storeLabels}}
```
Comment (tennix) [inline charts/tidb-cluster/templates/tidb-cluster.yaml]:
```suggestion
  {{- if .Values.tikv.storeLabels }}
```

File: charts/tidb-cluster/templates/tidb-cluster.yaml [hunk 3]
```diff
@@ -67,6 +67,16 @@ spec:
...
+    storeLabels:
+{{ toYaml .Values.tikv.storeLabels | indent 4 }}
+  {{- else }}
+    storeLabels:
+      - region
+      - zone
+      - rack
+      - host
+  {{- end}}
```
Comment (tennix) [inline charts/tidb-cluster/templates/tidb-cluster.yaml]:
```suggestion
  {{- end }}
```

File: charts/tidb-cluster/values.yaml [hunk 4]
```diff
@@ -192,6 +193,14 @@ tikv:
   #   effect: "NoSchedule"
   annotations: {}
 
+  ## storeLabels is used to define store label keys
```
Comment (tennix) [inline charts/tidb-cluster/values.yaml]: In order to use the location awareness feature of TiKV, users have to label their k8s nodes with the same labels.

File: pkg/manager/member/tikv_member_manager.go [hunk 5]
```diff
@@ -559,28 +559,25 @@ func (tkmm *tikvMemberManager) setStoreLabelsForTiKV(tc *v1alpha1.TidbCluster) (
...
+	labels := map[string]string{}
+	ls := node.GetLabels()
+	for _, storeLabel := range storeLabels {
+		if value, found := ls[storeLabel]; found {
+			labels[storeLabel] = value
+			continue
 		}
-		if host, found := ls[apis.LabelHostname]; found {
-			labels["host"] = host
+		if storeLabel == "host" {
```
Comment (tennix) [inline pkg/manager/member/tikv_member_manager.go]: Add a TODO note that this needs to be changed when PD supports label key containing slash character.

**Comment by weekface** [comment]
@tennix need another approval

**Pull request merged**